pandas/core/indexes/base.py

```diff
@@ -550,7 +550,13 @@ def __array_wrap__(self, result, context=None):
 
         attrs = self._get_attributes_dict()
         attrs = self._maybe_update_attributes(attrs)
-        return Index(result, **attrs)
+        from pandas.core.dtypes.generic import ABCDatetimeIndex
```

Comment:
not sure if you saw my comment. pls move this logic to ``_new_Index`` (and the import can be at the top of the file)